File: phalcon/app/web.php
<?php

use Phalcon\Di;
use Phalcon\Mvc\Router;
use Phalcon\Mvc\Dispatcher;
use Phalcon\Mvc\Application;
use app\Env;

define('BASE_PATH', dirname(__DIR__));
define('APP_PATH', BASE_PATH . '/app');

// Composer
require BASE_PATH.'/vendor/autoload.php';

try {
  // 注册堆栈框架
  $di = new Di();
  // 自动加载
  require APP_PATH.'/config/loader.php';
  // 公共服务
  require APP_PATH.'/config/services.php';
  /*  注册：路由 */
	$di->setShared('router', function () {
    $router = new Router();
    // 默认
		$router->setDefaultModule('home');
		return $router;
  });
  /* 注册：适配器 */
	$di->setShared('dispatcher', function() {
    $dispatcher = new Dispatcher();
    // 默认命名空间
		$dispatcher->setDefaultNamespace('app\modules\home\controller');
		return $dispatcher;
	});

  // 创建应用
	$app = new Application($di);
  // 注册：模块
	$app->registerModules([
		'home'=>[
			'className'=>'app\modules\home\module',
			'path'=>APP_PATH.'/modules/home/Module.php',
		],
		'admin'=>[
			'className'=>'app\modules\admin\module',
			'path'=>APP_PATH.'/modules/admin/Module.php',
		],
		'api'=>[
			'className'=>'app\modules\api\module',
			'path'=>APP_PATH.'/modules/api/Module.php',
		],
  ]);
  // 路由适配
  require APP_PATH . '/config/routes.php';
  // 执行
  $uri = explode('?', $di->get('request')->getURI());
	$app->handle($uri[0]??'/')->send();
}catch (\Exception $e){
  if(Env::$debug){
    echo $e->getMessage().'<br>';
    echo '<pre>'.$e->getTraceAsString().'</pre>';
  }else{
    header('Access-Control-Allow-Origin:*');
    header('Access-Control-Allow-Methods:*');
    header('Access-Control-Allow-Headers:Origin, X-Requested-With, Content-Type, Accept');
    echo json_encode(['code'=>5000,'msg'=>$e->getMessage()]);
  }
}

File: python/app/web.py
#!/bin/python
# -*- coding: UTF-8 -*-

import string
import json
from app.Env import Env
from flask import Flask,request,make_response
import flask_cors

# 框架
app = Flask(__name__)
app.debug=Env.debug
# 允许跨域
flask_cors.CORS(app)

# 执行
def _start(m,c,a,p=[]):
  try :
    c = string.capwords(c)
    exec('from app.modules.'+m+'.'+c+'Controller import *')
    exec('obj = '+c+'Controller()')
    return eval('obj.'+a+'('+(','.join(list(filter(None,p))))+')')
  except Exception as e :
    if Env.devinfo :
      pass
    else :
      res = make_response(json.dumps({'code':5000,'msg':'%s'%(e)}))
      res.status = '200'
      res.headers['Access-Control-Allow-Origin'] = "*"
      res.headers['Access-Control-Allow-Methods'] = "*"
      res.headers['Access-Control-Allow-Headers'] = "Origin, X-Requested-With, Content-Type, Accept"
      return res

# Home
@app.route('/',methods=['GET', 'POST'])
@app.route('/<c>',methods=['GET', 'POST'])
@app.route('/<c>/<a>',methods=['GET', 'POST'])
@app.route('/<c>/<a>/<p1>',methods=['GET', 'POST'])
@app.route('/<c>/<a>/<p1>/<p2>',methods=['GET', 'POST'])
@app.route('/<c>/<a>/<p1>/<p2>/<p3>',methods=['GET', 'POST'])
def index(c='Index',a='index',p1='',p2='',p3=''):
  # 排除路由
  url = request.path.split('/',2)[1].lower()
  if(url not in Env.exclude): return _start('home',c,a,[p1,p2,p3])
  else: return ''

# Api
@app.route('/api',methods=['GET', 'POST'])
@app.route('/api/<c>',methods=['GET', 'POST'])
@app.route('/api/<c>/<a>',methods=['GET', 'POST'])
@app.route('/api/<c>/<a>/<p1>',methods=['GET', 'POST'])
@app.route('/api/<c>/<a>/<p1>/<p2>',methods=['GET', 'POST'])
@app.route('/api/<c>/<a>/<p1>/<p2>/<p3>',methods=['GET', 'POST'])
def api(c='Index',a='index',p1='',p2='',p3=''):
  return _start('api',c,a,[p1,p2,p3])

# Admin
@app.route('/admin',methods=['GET', 'POST'])
@app.route('/admin/<c>',methods=['GET', 'POST'])
@app.route('/admin/<c>/<a>',methods=['GET', 'POST'])
@app.route('/admin/<c>/<a>/<p1>',methods=['GET', 'POST'])
@app.route('/admin/<c>/<a>/<p1>/<p2>',methods=['GET', 'POST'])
@app.route('/admin/<c>/<a>/<p1>/<p2>/<p3>',methods=['GET', 'POST'])
def admin(c='Index',a='index',p1='',p2='',p3=''):
  return _start('admin',c,a,[p1,p2,p3])

# 调试
if __name__ == '__main__':
  if(Env.debug): app.run(host='127.0.0.1',port=5000)
